integrasjonspunkt/src/main/java/no/difi/meldingsutveksling/spring/cloud/SpringCloudProtocolResolver.java

```diff
@@ -33,13 +36,24 @@
     public void initialize(final ConfigurableApplicationContext c) {
         logger.info("Initialize with Spring Cloud Config Binary resource resolver.");
         c.addProtocolResolver(new ProtocolResolver() {
+
+            Map<String, ByteArrayResource> maps = new HashMap<>();
+
             @Override
             public Resource resolve(String location, ResourceLoader rl) {
 
                 if (!location.startsWith(PREFIX)) {
                     return null;
                 }
 
+                if (!maps.containsKey(location)) {
+                    fetchRemote(location);
+                }
+
+                return maps.get(location);
+            }
+
+            private void fetchRemote(String location) throws BeansException {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove the declaration of thrown exception 'org.springframework.beans.BeansException' which is a runtime exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)